src/server/generic_family.cc

```diff
@@ -463,6 +458,164 @@ void GenericFamily::Stick(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(match_cnt);
 }
 
+struct SortEntryEmpty {};
+
+struct SortEntryScore {
+  double score;
+};
+
+template <
+    // Whether alphanumeric sort will be used.
+    bool ALPHA,
+    // Deduced StorageKind type for optimal size.
+    container_utils::StorageKind STORAGE_KIND,
+    // Whether to store double score. It will be stored for non-alphanumeric sorts
+    // when our items are not trivailly convertiable to doubles (i.e. they are not integers)
+    bool STORE_SCORE = !ALPHA && STORAGE_KIND != container_utils::StorageKind::INTEGRAL>
+struct SortEntry : public std::conditional_t<STORE_SCORE, SortEntryScore, SortEntryEmpty> {
```

Comment:
I initially tried this with more inheritance. But the logic quickly becomes scattered and it seems to me that `if constexpr()` chains are easier to understand.